### DISCUSSION TITLE
Naming Convention for Projects and Microservice

### DISCUSSION THREAD

**Discussion by rmuller-vu** [mia-platform/community#445] [Q&A]
### Description

I would like to know the naming convention the platform enforces when creating a new project and when naming a microservice. We would like to use `.` in the naming of a Microservice but the console doesn't allow it.

**Comment by davidebianchi** [comment]
Hi @rmuller-vu!

In the creation of a microservice, we adhere to the [DNS label](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names) convention (as recommended by Kubernetes). Since this name serves as both the pod name and hostname within Kubernetes, it may lead to unexpected behavior.

As for the project name, it should be human-readable. However, within the user interface (UI), we generate the projectId based on it, following the same rules as previously mentioned (as it's also utilized in namespace creation during environment setup).

Do you have any specific requirements?

**Reply by phil-hotchkiss** [reply]
I think we are fine with the DNS label convention for both the namespace and the name of the microservice. We would, however, like to use `.` in the name of the microservice repository. For example `mpms.auto-payoff-api` would be an example name for a microservice repository. The interface currently doesn't support that naming convention. Is there a particular justification for not wanting to use `.` in the name of the microservice repository when creating from a template?

**Reply by davidebianchi** [reply]
At the moment, the name of the repository is equal to the name of the service. So, it follows the same conventions above by default. It should be possible to modify the git repository name from the microservice creation page, have you tried it? Do you need some further automation

**Reply by phil-hotchkiss** [reply]
When modifying on this page, we get an error with regard to the naming requirements for the git repository. I believe clicking create from this page immediately creates the repository in git. We are able to use the naming standard we prefer when creating from a docker image, but that's only because the microservice has to already exist. As far as the ask, we're wondering if there is a way to change the regex used for the repository name or an alternative approach that still allows us to use our preferred naming convention while using the create from marketplace template option. 

![image](https://github.com/mia-platform/community/assets/136851567/12375693-7bff-407b-9fff-b381bda41eef)

**Reply by TuckerVaughan** [reply]
It's worth noting that when we tried to change the Git Repository name to differ from the name of microservice we got the same error we got when trying to change the name of the microservice.

![mia-being-cooperative_720](https://github.com/mia-platform/community/assets/130164234/bfd3ae4c-2479-41ff-992b-fa7eda307b0d)
![mia-git-name_720](https://github.com/mia-platform/community/assets/130164234/4bd6501f-e851-4704-83f1-425beca7379b)

**Reply by davidebianchi** [reply]
Thank you for the info! Sorry, I didn't remember the validation on the repository name field.
We could modify the `Git repository name` validation to accept also dots `.` in a future release. It would be ok for you?

**Reply by TuckerVaughan** [reply]
That would be great. Is there a chance that change can also be made for the microservice field name as well that way our microservice name matches our git repository name?

**Reply by phil-hotchkiss** [reply]
@davidebianchi I think that would be great to allow this for the microservice repository name. @TuckerVaughan I don't think that's necessary as the repository name is following the DNS label convention which doesn't allow for `.` characters to be used for dns except in subdomain names which the . in this case does not represent.

**Reply by TuckerVaughan** [reply]
@phil-hotchkiss Sounds good. Just changing the git repository name should suffice.
To add, should we mention the fact the name for the configuration repos that get generated for Mia projects have a different casing than the name for the Mia projects? For example, if I create a Mia project titled `RMS`, then it's configuration repo will be named `Rms-Configurations` instead of `RMS-Configurations`.
